### DISCUSSION TITLE
Is it possible to execute custom scripts (e.g. shell script) along with Kube manifests deployment

### DISCUSSION THREAD

**Discussion by Arpan-Chakra** [argoproj/argo-cd#5052] [Q&A]
How to run custom scripts (e.g. grrovy or shell script) using ArgoCD. Currently we use Jenkins and use JenkinsFile to run groovy commands and few shell scripts before running Kustomize to deploy. Kindly respond thanks!

**Comment by jannfis** [comment]
You might want to have a look at [custom config management plugins](https://argoproj.github.io/argo-cd/user-guide/config-management-plugins/). Be aware that you might have to build a custom Docker image containing all your required scripts, as described [here](https://argoproj.github.io/argo-cd/operator-manual/custom_tools/).

Also, when using a custom config management plugin, Kustomize will not be run automatically. So you would have to explicitly render out your manifests with Kustomize within your plugin. This in turn would disable some of the built-in Kustomize features, such as setting images or other properties from within the CLI or UI.

**Reply by Arpan-Chakra** [reply]
Thanks for responding!

I checked the above ArgoCd feature and found that this is more of Kubernetes way (configMap and plugins) of running scripts which can be complex and difficult to maintain. We have few shell scripts where we do some env variables check and set parameters and run kustomize e.g. (please check Attached scripts - deployEnv internally calls others) which we want to execute directly from source code. Please have a look and let me know if there is a direct way argo cd supports of running them. Cheers!


[deployEnv.sh.txt](https://github.com/argoproj/argo-cd/files/5692553/deployEnv.sh.txt)
[deployMeshConifg.sh.txt](https://github.com/argoproj/argo-cd/files/5692834/deployMeshConifg.sh.txt)
[deployPrometheus.sh.txt](https://github.com/argoproj/argo-cd/files/5692835/deployPrometheus.sh.txt)
[deployService.sh.txt](https://github.com/argoproj/argo-cd/files/5692836/deployService.sh.txt)

**Reply by Arpan-Chakra** [reply]
Yes and the above scripts use Kustomize pretty much, however with some extra checks.

**Comment by mlopez-eb** [comment]
3 years later and no response to this?

**Reply by crenshaw-dev** [reply]
[Jann's response](https://github.com/argoproj/argo-cd/discussions/5052#discussioncomment-204763) is correct - the supported way to do advanced config rendering, e.g. by running custom commands, is to use a Config Management Plugin.
